frontend/src/metabase/setup/components/SettingsPage/SettingsPage.tsx

```diff
@@ -30,13 +32,22 @@ const STEP_COMPONENTS: Partial<
 
 export const SettingsPage = (): JSX.Element => {
   const steps = useSelector(getSteps);
+  const isEmbeddingUseCase = useSelector(getIsEmbeddingUseCase);
+  const SELECT_WIDTH = 160;
 
   return (
     <div data-testid="setup-forms">
-      <PageHeader>
-        <LogoIcon height={51} />
-      </PageHeader>
-      <PageBody>
+      <Box component="header" className={S.PageHeader}>
+        <Flex
+          align="center"
+          justify={isEmbeddingUseCase ? "space-between" : "center"}
+        >
+          {isEmbeddingUseCase && <Box w={SELECT_WIDTH} className={S.Decoy} />}
+          <LogoIcon height={51} />
+          {isEmbeddingUseCase && <LanguageSelector w={SELECT_WIDTH} />}
+        </Flex>
+      </Box>
```

Comment:
nit, but this markup is a little simpler: Always display 3 sections, but only populate the one on the right in the embedding use case. This lets you have one conditional instead of 3, and you don't have to pass a prop to `LanguageSelector` since it's contained by the box

```tsx
<Flex
  align="center"
  justify="space-between"
>
  <Box w={SELECT_WIDTH} className={S.Decoy} />
  <LogoIcon height={51} />
  <Box w={SELECT_WIDTH}>
    {isEmbeddingUseCase && <LanguageSelector />}
  </Box>
</Flex>
```

Reply:
Great thinking!

Reply:
https://github.com/metabase/metabase/pull/57749/commits/9e1673eca0d08242bbbf8e6e4bfee587a164b575